src/RamIndexAnalysis.cpp

```diff
@@ -387,8 +387,10 @@ SearchSignature RamIndexAnalysis::getSearchSignature(
         const RamProvenanceExistenceCheck* provExistCheck) const {
     const auto values = provExistCheck->getValues();
     SearchSignature res = 0;
-    // values.size() - 1 because we discard the height annotation
-    for (int i = 0; i < (int)values.size() - 1; i++) {
+    auto numberOfHeights = provExistCheck->getRelation().getNumberOfHeights();
+    // values.size() - numberOfHeights because we discard the height annotations
+
+    for (int i = 0; i < (int)values.size() - (int)numberOfHeights; i++) {
```

Comment:
Can you just use `size_t i` to avoid the casting

Reply:
Incorporated in clean up commit.